python/amici/sbml_import.py

```diff
@@ -1473,6 +1468,40 @@ def _parse_special_functions(sym: sp.Expr, toplevel: bool = True) -> sp.Expr:
     return sym
 
 
+def _parse_piecewise_to_heaviside(args: Iterable[sp.Expr]) -> sp.Expr:
+    """
+   Piecewise functions cannot be transformed into C++ right away, but AMICI has
+   a special interface for Heaviside function, so we transform them
+
+    :param args:
+        symbolic expressions for arguments of the piecewise function
+    """
+    # how many condition-expression pairs will we have?
+    formula = sp.sympify('0')
+    lastroot = sp.Float(1.0)
+    conditions = list(grouper(args, 2, False))
+    for coeff, root in conditions:
+        if root != sp.false:
+            # Sympy doesn't assume by default that the expression is real...
+            root._assumptions['extended_real'] = True
```

Comment:
```suggestion
            root._assumptions['extended_real'] = True
```

Is there a cleaner way for modifying assumptions. Checked sometime ago, but didn't find any. Maybe somebody else knows.

Reply:
I know what you mean. I looked into this a little, but didn't find a good API doc for this... If there is one, it's not well findable. As the current implementation works and is also clear about it's purpose (although not super nice, as opeating on protected variables), I wouldn't bother with searching longer than a few minutes for an API...